slam-libraries/viam-cartographer/src/slam_service/slam_service.cc

```diff
@@ -194,6 +194,17 @@ ::grpc::Status SLAMServiceImpl::GetPointCloudMap(const GetMapRequest *request,
     }
 }
 
+::grpc::Status SLAMServiceImpl::GetInternalState(
+    ServerContext *context, const GetInternalStateRequest *request,
+    GetInternalStateResponse *response) {
+    {
+        std::lock_guard<std::mutex> lk(map_builder_mutex);
+        std::string buf = map_builder.SaveMapToStream(path_to_map);
+        response->set_internal_state(buf);
```

Comment:
This part doesn't need to be done under the mutex.

Reply:
Why does this not need to be locked? aren't lidar scans still being added here? Or are the proto methods safe?

Reply:
nevermind, just realized the lock scopes the entire grpc call, which is not what I intended when first suggesting a lock. Also missed how much the code changed since then

Reply:
Right, we should call SaveMapToStream under the lock, but we don't need to call set_internal_state under the lock.

Reply:
Reducing the scope of the mutex even further by only doing the SaveFile in the mutex, not the reading from file. We are using unique UUIDs for that.